e3sm_diags/derivations/formulas.py

```diff
@@ -72,12 +74,29 @@ def w_convert_q(var: xr.DataArray):
     return var
 
 
-def molec_convert_units(var: xr.DataArray, molar_weight: float):
+def molec_convert_units(vars: List[xr.DataArray], molar_weight: float) -> xr.DataArray:
+    """Sum the list of variables and convert the molecular units.
+
+    Parameters
+    ----------
+    vars : List[xr.DataArray]
+        The list of variables.
+    molar_weight : float
+        The molar weight to use for the conversion.
+
+    Returns
+    -------
+    xr.DataArray
+        The final result.
+    """
+    result = sum_vars(vars)
+
     # Convert molec/cm2/s to kg/m2/s
-    if var.attrs["units"] == "molec/cm2/s":
-        var = var / AVOGADRO_CONST * molar_weight * 10.0
-        var.attrs["units"] == "kg/m2/s"
-    return var
+    if result.attrs["units"] == "molec/cm2/s":
+        result = result / AVOGADRO_CONST * molar_weight * 10.0
+        result.attrs["units"] = "kg/m2/s"
+
+    return result
 
```

Comment:
This function now expects a list of variables (`xr.DataArray`), instead of a single sum DataArray. 
We need to pass the list of variables to the `sum_vars` function to maintain the `"units"` attribute.

Otherwise, calling `sum(vars)` beforehand will drop attributes.